tools/filevanish.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# @lint-avoid-python-3-compatibility-imports
+#
+# filevanish    Trace why file vanished
+#             For Linux, uses BCC, eBPF. Embedded C.
+#
+# USAGE: filevanish [-h] [-p PID]
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# 08-Nov-2022 Curu. modified from filelife
+
+from __future__ import print_function
+from bcc import BPF
+import argparse
+from time import strftime
+
+# arguments
+examples = """examples:
+    ./filevanish           # trace all stat() syscalls
+    ./filevanish -p 181    # only trace PID 181
+"""
+parser = argparse.ArgumentParser(
+    description="Trace stat() syscalls",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=examples)
+parser.add_argument("-p", "--pid",
+    help="trace this PID only")
+parser.add_argument("--ebpf", action="store_true",
+    help=argparse.SUPPRESS)
+args = parser.parse_args()
+debug = 0
+
+# define BPF program
+bpf_text = """
+#include <uapi/linux/ptrace.h>
+#include <linux/fs.h>
+#include <linux/sched.h>
+
+struct data_t {
+    u32 pid;
+    u8 action;
+    char comm[TASK_COMM_LEN];
+    char fname[DNAME_INLINE_LEN];
+    char fname2[DNAME_INLINE_LEN];
+};
+
+BPF_HASH(birth, struct dentry *);
+BPF_PERF_OUTPUT(events);
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 12, 0)
```

Comment:
We could use BTF info to avoid this version check, see #4339.

Reply:
maybe change it later.
currently we need to keep compatibility with bcc versions below v0.23 which has no 'kernel_struct_has_field'

Reply:
Hi, can we do this https://github.com/iovisor/bcc/pull/4346 ?

Reply:
> Hi, can we do this #4346 ?
RHEL/CentOS 7 comes with bcc version 0.10, which has no **kernel_struct_has_field**,   so the same script woud not run without modification.

so maybe we should drop support for that old version and use a legacy branch to do that?